File: client/src/types/something.ts
export type Something = { // TODO: init dynamic data
  id: number;
  deviceName: string;
  syncDeviceJoined: string;
  currentDevice: {};
  lastModifiedAt: number;
  type: 'recv' | 'self';
  appUsername: string;
  appEmailAddress: string;
  appAlignedCb: boolean;
  appBillingPeriod: number;
  appSalary: number;
};

File: server/app/device/events/ws/something.go
package ws

type Something struct { // TODO: init dynamic data
	Id               string `json:"id"`
	Username         string `json:"username"`
	DeviceName       string `json:"deviceName"`
	SyncDeviceJoined string `json:"syncDeviceJoined"`
	CurrentDevice    Device `json:"currentDevice"`
	LastModifiedAt   int    `json:"lastModifiedAt"`
	AppUsername      string `json:"appUsername"`
	AppEmailAddress  string `json:"appEmailAddress"`
	AppAlignedCb     bool   `json:"appAlignedCb"`
	AppBillingPeriod int    `json:"appBillingPeriod"`
	AppSalary        int    `json:"appSalary"`
}

type Device struct {
	Name    string `json:"name"`
	Version string `json:"version"`
}
